### PR TITLE
Fallback to system-wide metrics under cgroups

### LINKED ISSUE #71733: 24.9+ some OS-level metrics are now not available in containers (or if cgroups are used)
The system-wide metrics like LoadAverage1 OSProcessesRunning OSInterrupts etc. are not available after that change https://github.com/ClickHouse/ClickHouse/pull/62003

There are no cgroup-local analogue for LoadAverage, so IMHO it should stay (and it's ok that they are system-wide, should be just clearly stated in the metric description).

And if we talk about cgroup-specific metrics 'nice to have' but still not there:

/sys/fs/cgroup/<cgroup_name>/cpu.pressure
/sys/fs/cgroup/<cgroup_name>/memory.pressure
/sys/fs/cgroup/<cgroup_name>/io.pressure

(see https://docs.kernel.org/accounting/psi.html )

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
When ClickHouse runs under cgroup we will still collect system-wide asynchronous metrics related to system load, process scheduling, memory etc. They might provide useful signals when ClickHouse is the only process on the host with high resource consumption. 

---

Closes: #71733

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 41d89d2c94639b403d4030fd05b4276f0541520f with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/73369/41d89d2c94639b403d4030fd05b4276f0541520f/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/73369/41d89d2c94639b403d4030fd05b4276f0541520f/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

File: src/Common/AsynchronousMetrics.cpp [hunk 1]
```diff
@@ -1049,6 +1043,14 @@ void AsynchronousMetrics::update(TimePoint update_time, bool force_update)
 
                 if (name.starts_with("cpu"))
                 {
+                    if (cgroup_cpu_metrics_present)
```
Comment (devcrafter) [inline src/Common/AsynchronousMetrics.cpp:1046]: Wouldn't it be simpler always to read `ProcStatValuesCPU` first (it doesn't depend on the number of CPU, so the check below is irrelevant)
```suggestion
    ProcStatValuesCPU current_values{};
    current_values.read(*proc_stat);
    
    /// Skip the CPU metrics if we already have them from cgroup
    if (cgroup_cpu_metrics_present)
        continue;
```

**Review by devcrafter — APPROVED** [review]
LGTM with comment

**Comment by nickitat** [comment]
`Stateless tests (tsan, s3 storage) [3/3]` - #73433

File: src/Common/AsynchronousMetrics.cpp [hunk 2]
```diff
@@ -1135,7 +1137,7 @@ void AsynchronousMetrics::update(TimePoint update_time, bool force_update)
 
                 Float64 num_cpus_to_normalize = max_cpu_cgroups > 0 ? max_cpu_cgroups : num_cpus;
 
-                if (num_cpus_to_normalize > 0)
+                if (num_cpus_to_normalize > 0 && !cgroup_cpu_metrics_present)
```
Comment (filimonov) [inline src/Common/AsynchronousMetrics.cpp:1140]: why? :)

Comment (nickitat) [reply src/Common/AsynchronousMetrics.cpp:1140]: have you saw cpu normalized not reported on a cgroups v1 host?

Comment (filimonov) [reply src/Common/AsynchronousMetrics.cpp:1140]: I didn't check if it was v1 or v2. But they are not there for sure: https://github.com/ClickHouse/ClickHouse/issues/76810  (there is even a simple test showing it it fiddle).

**Pull request merged**